include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -602,13 +603,13 @@ namespace CUDAWrappers
            cells_per_block_shmem(int dim, int fe_degree)
   {
     /* clang-format off */
-    return dim==2 ? (fe_degree==1 ? 32 :
-                     fe_degree==2 ? 8 :
-                     fe_degree==3 ? 4 :
-                     fe_degree==4 ? 4 :
+    return dim==2 ? (fe_degree==1 ? CUDAWrappers::warp_size :
+                     fe_degree==2 ? CUDAWrappers::warp_size/4 :
+                     fe_degree==3 ? CUDAWrappers::warp_size/8 :
+                     fe_degree==4 ? CUDAWrappers::warp_size/8 :
```

Comment:
Let't put here some comments:

```suggestion
    // we are trying to limit the number of threads according to the 
    // following formula:
    //    `threads = cells * (k+1)^d <= 4*CUDAWrappers::warp_size`
    return dim==2 ? (fe_degree==1 ? CUDAWrappers::warp_size :    // 128 <= 128 (for CUDA)
                     fe_degree==2 ? CUDAWrappers::warp_size/4 :  //  72 <= 128
                     fe_degree==3 ? CUDAWrappers::warp_size/8 :  //  64 <= 128
                     fe_degree==4 ? CUDAWrappers::warp_size/8 :  // 100 <= 128
```

Also, if @masterleinad and you have created an investigation regarding optimal number of cells maybe create an issue with those results and add a link here to that issue, so that we remember the reason for these (new?) numbers.

Reply:
These are the same numbers than before. I plan to revisit them when I start running real tests on AMD gpu.